src/fairseq2/nn/utils/module.py

```diff
@@ -97,20 +130,20 @@ def empty_like(src: Tensor) -> Tensor:
         return tgt
 
     def to_empty_(m: Module) -> None:
-        for name, prm in m.named_parameters(recurse=False):
-            if prm is None:
+        for name, param in m.named_parameters(recurse=False):
+            if param is None:
                 continue
 
             with torch.no_grad():
-                new_prm = Parameter(empty_like(prm), prm.requires_grad)
+                new_param = Parameter(empty_like(param), param.requires_grad)
 
-            setattr(m, name, new_prm)
+            setattr(m, name, new_param)
```

Comment:
Curious what is the benefit of using `setattr(m, name, new_param)` over `m.name = new_param`

Reply:
`name` here is of type string. So syntactically you cannot write a valid Python statement like `m."foo" = "xyz"`.

Reply:
got it - thank you :)

Reply:
no problem! :)